CRM/Mailing/Event/BAO/Subscribe.php

```diff
@@ -211,6 +211,8 @@ public function send_confirm_request($email) {
         )
       ) . "@$emailDomain";
 
+    $bounce_address = $localpart . "@$emailDomain";
```

Comment:
This will not result in bounce-processing by CiviCRM. I'd suggest instead:

```php
$bounce_address = implode($config->verpSeparator,
        array(
          $localpart . 'b',
          $this->contact_id,
          $this->id,
          $this->hash,
        )
      ) . "@$emailDomain";
```